### PR TITLE
Use snapkit to layout routeOptionsVC

### PR DESCRIPTION
**Pull request by Omarrasheed**
As the title says. This tackles routeOptionsVC and its extensions file

### CODE REVIEW

File: TCAT/Controllers/RouteOptionsViewController+Extensions.swift [hunk 1]
```diff
@@ -366,6 +361,7 @@ extension RouteOptionsViewController: DZNEmptyDataSetSource, DZNEmptyDataSetDele
         titleLabel.snp.makeConstraints { (make) in
             make.top.equalTo(symbolView.snp.bottom).offset(10)
             make.centerX.equalTo(symbolView.snp.centerX)
+            make.size.equalTo(titleLabel.intrinsicContentSize)
```
Comment (kevinchan159) [inline TCAT/Controllers/RouteOptionsViewController+Extensions.swift]: shouldnt need this

File: TCAT/Controllers/RouteOptionsViewController.swift [hunk 2]
```diff
@@ -33,11 +33,11 @@ enum RequestAction {
...
-    var datePickerOverlay: UIView!
-    var datePickerView: DatePickerView!
-    var routeResults: UITableView!
-    var routeSelection: RouteSelectionView!
-    var searchBarView: SearchBarView?
+    let datePickerOverlay = UIView()
+    let datePickerView = DatePickerView()
+    let routeResults = UITableView(frame: .zero, style: .grouped)
+    let routeSelection = RouteSelectionView()
+    let searchBarView = SearchBarView()
```
Comment (kevinchan159) [inline TCAT/Controllers/RouteOptionsViewController.swift:40]: make `private`

File: TCAT/Controllers/RouteOptionsViewController.swift [hunk 3]
```diff
@@ -111,11 +103,12 @@ class RouteOptionsViewController: UIViewController {
             registerForPreviewing(with: self, sourceView: view)
         }
 
+        setupConstraints()
+
```
Comment (kevinchan159) [inline TCAT/Controllers/RouteOptionsViewController.swift]: delete newline

File: TCAT/Controllers/RouteOptionsViewController.swift [hunk 4]
```diff
@@ -188,18 +177,76 @@ class RouteOptionsViewController: UIViewController {
...
     }
 
+    private func setupDatePicker() {
+        setupDatePickerView()
+        setupDatePickerOverlay()
+    }
+
+    private func setupDatePickerView() {
+        datePickerView.delegate = self
+
```
Comment (kevinchan159) [inline TCAT/Controllers/RouteOptionsViewController.swift]: delete newline

File: TCAT/Controllers/RouteOptionsViewController.swift [hunk 5]
```diff
@@ -188,18 +177,76 @@ class RouteOptionsViewController: UIViewController {
...
+        view.addSubview(datePickerView) // so datePicker can go ontop of other views
+    }
+
+    private func setupDatePickerOverlay() {
+        datePickerOverlay.backgroundColor = Colors.black
+        datePickerOverlay.alpha = 0
+        datePickerOverlay.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(self.dismissDatePicker)))
+
```
Comment (kevinchan159) [inline TCAT/Controllers/RouteOptionsViewController.swift]: delete newline

File: TCAT/Controllers/RouteOptionsViewController.swift [hunk 6]
```diff
@@ -188,18 +177,76 @@ class RouteOptionsViewController: UIViewController {
...
+        }
+    }
+
+    func setupConstraintsForHiddenDatePickerView() {
+        self.datePickerView.snp.remakeConstraints { make in
+            make.top.equalTo(self.view.snp.bottom)
+            make.leading.trailing.equalToSuperview()
+            make.height.equalTo(self.datePickerView.frame.height)
+        }
+    }
```
Comment (kevinchan159) [inline TCAT/Controllers/RouteOptionsViewController.swift:222]: prefer to get rid of `self` if unnnecessary. same for lines 212-215

File: TCAT/Views/DatePickerView.swift [hunk 7]
```diff
@@ -177,6 +178,7 @@ class DatePickerView: UIView {
         datepicker.snp.makeConstraints { make in
             make.leading.trailing.equalToSuperview()
             make.top.equalTo(leaveNowSegmentedControl.snp.bottom).offset(spaceBtSegmentControlAndDatePicker)
+            make.height.equalTo(164.5)
```
Comment (kevinchan159) [inline TCAT/Views/DatePickerView.swift]: make constant

**Review by kevinchan159 — CHANGES_REQUESTED** [review]
amazing!!! 
just some minor comments

File: TCAT/Controllers/RouteOptionsViewController.swift (hunk 2 above)
Comment (Omarrasheed) [reply TCAT/Controllers/RouteOptionsViewController.swift:40]: All of these end up getting accessed in the extensions file. I'll try to reorganize these when tackling #251 and #252

**Review by kevinchan159 — APPROVED** [review]

**Pull request merged**